src/test_report.py

```diff
@@ -96,32 +95,49 @@ def send_mail(self, email_msg, email_server):
         else:
             self._logger.log_message(logging.INFO, "Email Sent.")
 
+    def get_test_analysis(self, nodes):
+        total_runs = len(nodes)
+        total_failures = sum(node['status'] == "fail" for node in nodes)
+        total_timeouts = sum(node['status'] == "timeout" for node in nodes)
+        return total_runs, total_failures, total_timeouts
+
+    def create_test_report(self, root_node, child_nodes):
+        total_runs, total_failures, total_timeouts = \
+                    self.get_test_analysis(child_nodes)
+
+        template_env = jinja2.Environment(
+                            loader=jinja2.FileSystemLoader("./config/reports/")
+                        )
+        template = template_env.get_template("test-report.jinja2")
+        subject_str = (f"{root_node['revision']['tree']}/\
+{root_node['revision']['branch']} \
+{root_node['revision']['describe']}: \
+{total_runs} runs {total_failures} fails \
+({root_node['name']})")
+        email_content = template.render(subject_str=subject_str,
+                                        root=root_node,
+                                        tests=child_nodes)
+        return email_content, subject_str
+
     def run(self):
-        sub_id = self._db.subscribe('node')
+        sub_id = self._db.subscribe_node_channel(filters={
+            'op': 'completed',
```

Comment:
This makes me think, `op` should be renamed as it was originally about an `operation` like creating or updating a node but now the value passed is a status. I think the schema of the data passed when subscribing will need to be improved too to deal with the changes in the models.

Reply:
Yeah I don't think it's something to do as part of this PR, it was just a comment for something to do later on.